packages/api/src/command/hie/unlink-patient-from-organization.ts

```diff
@@ -0,0 +1,345 @@
+    }
+  } catch (error) {
+    log("Error removing medical document from S3:");
+    throw error;
+  }
+}
+
+async function findAndRemoveConsolidatedDocumentFromS3(
+  cxId: string,
+  patientId: string,
+  dryRun = false,
+  log: typeof console.log
+): Promise<void> {
+  const dryRunMsg = getDryRunPrefix(dryRun);
+  try {
+    const consolidatedPrefix = createConsolidatedSnapshotFileNameWithNoExtension(cxId, patientId);
+    const medicalRecordsPrefix = createMRSummaryFileNameWithNoExtension(cxId, patientId);
+
+    const [existingConsolidatedFiles, existingMedicalRecordsFiles] = await Promise.all([
+      s3Utils.listObjects(s3MedicalDocumentsBucketName, consolidatedPrefix),
+      s3Utils.listObjects(s3MedicalDocumentsBucketName, medicalRecordsPrefix),
+    ]);
+
+    const existingFilenames: string[] = [];
+
+    if (existingConsolidatedFiles) {
+      const consolidatedFileNames = existingConsolidatedFiles.flatMap(file => file.Key ?? []);
+      existingFilenames.push(...consolidatedFileNames);
+    }
+
+    if (existingMedicalRecordsFiles) {
+      const medicalRecordsFileNames = existingMedicalRecordsFiles.flatMap(file => file.Key ?? []);
+      existingFilenames.push(...medicalRecordsFileNames);
+    }
+
+    if (existingFilenames.length > 0) {
+      log(`${dryRunMsg}Deleting ${existingFilenames.length} files from S3`);
+
+      if (!dryRun) {
+        await s3Utils.deleteFiles({
+          bucket: s3MedicalDocumentsBucketName,
+          keys: existingFilenames,
+        });
+      }
+    }
+  } catch (error) {
+    log("Error removing consolidated documents from S3:");
+    throw error;
+  }
+}
+
+async function findAndInvalidateLinks(
+  cwLink: CwLink | undefined,
+  cqLink: CQLink | undefined,
+  cxId: string,
+  patientId: string,
+  dryRun = false,
+  log: typeof console.log
+): Promise<void> {
+  const dryRunMsg = getDryRunPrefix(dryRun);
+  try {
+    const invalidLinks = {
+      carequality: cqLink ? [cqLink] : [],
+      commonwell: cwLink ? [cwLink] : [],
+    };
+
+    if (dryRun) {
+      log(`${dryRunMsg}Would invalidate links:`, invalidLinks);
+      return;
+    }
+
+    await Promise.all([
+      createOrUpdateInvalidLinks({ id: patientId, cxId, invalidLinks }),
+      updateCQPatientData({ id: patientId, cxId, invalidateLinks: invalidLinks.carequality }),
+      updateCwPatientData({ id: patientId, cxId, invalidateLinks: invalidLinks.commonwell }),
+    ]);
+  } catch (error) {
+    log("Error invalidating links:");
+    throw error;
+  }
+}
+
+async function deleteFromOpenSearch(entryId: string, dryRun = false, log: typeof console.log) {
+  const dryRunMsg = getDryRunPrefix(dryRun);
+  const openSearch = makeSearchServiceRemover();
+  try {
+    if (!dryRun) {
+      await openSearch.remove(entryId);
+    } else {
+      log(`${dryRunMsg}Would delete entry ${entryId} from OpenSearch`);
+    }
+  } catch (error) {
+    log("Error deleting from OpenSearch:", error);
```

Comment:
ditto: pass the string representation of the error inside the message, avoid multi-line logs